pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -783,6 +796,19 @@ func (c *Cluster) getActivateOSDInitContainer(configDir, namespace, osdID string
 	return volume, container
 }
 
+func (c *Cluster) getnewkeyring(osdId string, osdProps osdProperties) v1.Container {
```

Comment:
Instead of creating a new init container, I wonder if we could include this in the existing `activate` init container script.

Reply:
+1 for the activate init container which has all the bits and pieces to run ceph commands already.

Reply:
I'm getting `[errno 13] RADOS permission denied (error connecting to the cluster)` when I check the log.
I tried `ReadOnly: false` in the volume mount but that didn't work I think `'allow *'` will require `rwx`

Reply:
You won't need all of that if you use the activate container like suggested above.

Reply:
I have the command in activate init container only and I'm seeing 
```
ceph-volume raw activate successful for osd ID: 0
+ ceph auth get-or-create osd.0 mon 'profile osd' mgr 'profile osd' osd 'allow *'
2021-05-24T15:30:37.222+0000 7f143e7ec700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-05-24T15:30:37.222+0000 7f143e7ec700 -1 AuthRegistry(0x7f143805e620) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-05-24T15:30:37.224+0000 7f143e7ec700 -1 auth: unable to find a keyring on /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,: (2) No such file or directory
2021-05-24T15:30:37.224+0000 7f143e7ec700 -1 AuthRegistry(0x7f143e7eaea0) no keyring found at /etc/ceph/ceph.client.admin.keyring,/etc/ceph/ceph.keyring,/etc/ceph/keyring,/etc/ceph/keyring.bin,, disabling cephx
2021-05-24T15:30:37.224+0000 7f1437fff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [1]
2021-05-24T15:30:37.224+0000 7f143e7ec700 -1 monclient: authenticate NOTE: no keyring found; disabled cephx authentication
[errno 13] RADOS permission denied (error connecting to the cluster)
```